lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -553,10 +600,12 @@ def create_route_spec(connection, module, vpc_id):
         rename_key(route_spec, 'dest', 'destination_cidr_block')
 
         if route_spec.get('gateway_id') and route_spec['gateway_id'].lower() == 'igw':
```

Comment:
While testing I noticed this doesn't even get triggered because route_spec looks something like
`{'gateway_id': 'igw-xxxxxxxx', 'destination_cidr_block': '0.0.0.0/0'}`. Should probably be `... and route_spec['gateway_id'].lower().startswith('igw'):`

Reply:
The next step if this condition matches is to find the IGW. This is tested for in line 109 of the test suite.